tests/Doctrine/Tests/ORM/Functional/BasicFunctionalTest.php

```diff
@@ -258,6 +256,31 @@ public function testBasicQuery(): void
         self::assertEquals('developer', $usersScalar[0]['u_status']);
     }
 
+    public function testSingleColumnQuery(): void
+    {
+        $gregoire           = new CmsUser();
+        $gregoire->name     = 'Gregoire';
+        $gregoire->username = 'greg0ire';
+        $gregoire->status   = 'developer';
+        $this->_em->persist($gregoire);
+
+        $bhushan           = new CmsUser();
+        $bhushan->name     = 'Bhushan';
+        $bhushan->username = 'bhushan';
+        $bhushan->status   = 'developer';
+        $this->_em->persist($bhushan);
+
+        $this->_em->flush();
+
+        $query = $this->_em->createQuery('select u.username from Doctrine\Tests\Models\CMS\CmsUser u');
+
+        $users = $query->getSingleColumnResult();
+
+        self::assertCount(2, $users);
+        self::assertTrue(in_array('bhushan', $users));
```

Comment:
Only `postgres` has different order of these values, so changed it to use `in_array` for assertion.

cc @greg0ire

Reply:
@SenseException 

It is not actually related to this functionality,

```php
$gregoire           = new CmsUser();
        $gregoire->name     = 'Gregoire';
        $gregoire->username = 'greg0ire';
        $gregoire->status   = 'developer';
        $this->_em->persist($gregoire);

        $bhushan           = new CmsUser();
        $bhushan->name     = 'Bhushan';
        $bhushan->username = 'bhushan';
        $bhushan->status   = 'developer';
        $this->_em->persist($bhushan);
```

These lines create records `['bhushan', 'greg0ire']` when MySQLDriver and SqlDriver are used but when we use `PostgresDriver` records are stored `['greg0ire', 'bhushan']` like this which breaks tests.

May be its a bug or feature of PostgresDriver, I am not sure about this, You know better than me about this.

Regarding original question, (Please correct me if I am wrong) Hydrator is used only after query execution to fill up data, so order by should also work, Let me write test for it.

Reply:
Test added

Reply:
Tests are added and its working fine with order by clause. 

Do you think its a bug 🐛 in dbal which we should report ? 

Cc @greg0ire @SenseException

Reply:
I think if you don't specify an order, you should an assume any. In fact, you should assume that an implementation could return one order or the other on subsequent calls. In practice, it's very unlikely that they would. See for instance that Stack Overflow question: https://stackoverflow.com/a/10064571/353612

Reply:
So `testSingleColumnQuery()` still needs an order to compare it with an expected array.

Reply:
Should we mark this as resolved ?

Reply:
A hydrator is responsible for a resulting data structure. If it returns an array with assoc arrays as elements or an an array of objects or one array with elements of one column. This is what the tests should cover.

Usually a `select id from users` would be sufficient as test-query for its simplicity, but the mentioned different results between databases need to introduce a different query where the result is consistent. You also could use a much more complex query, but that would be unnecessary for a test case. That is why DESC and ASC (and neither of both) are actually testing the same case for a hydrator and its data structure. The order does not matter for the hydrator data structure. It was just added for multi platform compatibility.

From a documentation perspective of a test and what is expected from its result

``` php
$expected = [
    'bhushan',
    'greg0ire',
];
```

would be easier to read.

Reply:
removed order by tests and updated `testSingleColumnQuery` this test to use `DESC` orders for assertions